mobify/platform-scaffold
# Feature/mbfycnct-60-61 put-delete promo code

This PR covers functionality regarding the putPromoCode, deletePromoCode and initCartPage commands

 **JIRA**: ([MBFYCNCT-60](https://thinkwrap.jira.com/browse/MBFYCNCT-60)), ([MBFYCNCT-61](https://thinkwrap.jira.com/browse/MBFYCNCT-61)), ([MBFYCNCT-31](https://thinkwrap.jira.com/browse/MBFYCNCT-31))

## Changes
-  Added putPromoCode command
-  Added deletePromoCode command
-  Added initCartPage command
-  Added getCartTotals() to cart/utils.js
-  Added populateLocationsData to checkout/utils.js
-  Added parseDeliveryCountries to checkout/parsers.js

## Todos
- [ ] Review discount amount displayed in cart page

## How to test-drive this PR
- Run `npm run dev`
- Preview [Thinkwrap accelerator](https://preview.mobify.com/?url=https%3A%2F%2Fhydemo-electronics.thinkwrap.com%2Fyacceleratorstorefront%2Felectronics%2Fen%2F%2F&site_folder=https%3A%2F%2Flocalhost%3A8443%2Floader.js&disabled=0&domain=&scope=1)
- Login with user: alistair@hybris.com and password: 123456
- Go to cart page and check if cart has coupon applied
- If coupon is applied, delete it
- Apply coupon ‘CHRISTMAS16’
- Delete coupon
- Check, using Redux dev tools, that countries and regions are stored in the app state in checkout > locations. Countries are retrieved form /deliverycountries endpoint and regions from the connector configuration object.
